cmd/juju/storage/volumelist.go

```diff
@@ -0,0 +1,176 @@
+	Assigned    bool   `yaml:"assigned" json:"assigned"`
+	MachineId   string `yaml:"machine" json:"machine"`
+	Attached    bool   `yaml:"attached" json:"attached"`
+	DeviceName  string `yaml:"device-name" json:"device-name"`
+	UUID        string `yaml:"uuid" json:"uuid"`
+	Label       string `yaml:"label" json:"label"`
+	Size        uint64 `yaml:"size" json:"size"`
+	InUse       bool   `yaml:"in-use" json:"in-use"`
+	FSType      string `yaml:"file-system-type" json:"file-system-type"`
+	Provisioned bool   `yaml:"provisioned" json:"provisioned"`
+}
+
+// Run implements Command.Run.
+func (c *VolumeListCommand) Run(ctx *cmd.Context) (err error) {
+	api, err := getVolumeListAPI(c)
+	if err != nil {
+		return err
+	}
+	defer api.Close()
+
+	result, err := api.ListVolumes(c.Ids)
+	if err != nil {
+		return err
+	}
+	output := c.convertFromAPIDisks(result)
+	return c.out.Write(ctx, output)
+}
+
+var (
+	getVolumeListAPI = (*VolumeListCommand).getVolumeListAPI
+)
+
+// VolumeListAPI defines the API methods that the volume list command use.
+type VolumeListAPI interface {
+	Close() error
+	ListVolumes(machines []string) ([]params.StorageDisk, error)
+}
+
+func (c *VolumeListCommand) getVolumeListAPI() (VolumeListAPI, error) {
+	return c.NewStorageAPI()
+}
+
+func (c *VolumeListCommand) convertFromAPIDisks(all []params.StorageDisk) []DiskInfo {
+	result := make([]DiskInfo, len(all))
+	for i, one := range all {
+		result[i] = DiskInfo{
+			Attachments: c.convertFromAPIAttachments(one.Attachments),
+		}
+	}
+	return result
+}
+
+func (c *VolumeListCommand) convertFromAPIAttachments(all []params.StorageAttachment) []AttachmentInfo {
+	result := make([]AttachmentInfo, len(all))
+	for i, one := range all {
+		result[i] = AttachmentInfo{
+			Tag:         one.Tag,
+			StorageId:   one.StorageId,
+			Assigned:    one.Assigned,
+			MachineId:   one.MachineId,
+			Attached:    one.Attached,
+			DeviceName:  one.DeviceName,
+			UUID:        one.UUID,
+			Label:       one.Label,
+			Size:        one.Size,
+			InUse:       one.InUse,
+			FSType:      one.FSType,
+			Provisioned: one.Provisioned,
+		}
+	}
+	return result
+}
+
+func (c *VolumeListCommand) formatTabular(value interface{}) ([]byte, error) {
+	disks, valueConverted := value.([]DiskInfo)
+	if !valueConverted {
+		return nil, errors.Errorf("expected value of type %T, got %T", disks, value)
+	}
+	var out bytes.Buffer
+	const (
+		// To format things into columns.
+		minwidth = 0
+		tabwidth = 1
+		padding  = 2
+		padchar  = ' '
+		flags    = 0
+	)
+	tw := tabwriter.NewWriter(&out, minwidth, tabwidth, padding, padchar, flags)
+	fmt.Fprintf(tw, "TAG\tATTACHED\tMACHINE\tDEVICE NAME\tUUID\tLABEL\tSIZE\n")
+	for _, disk := range disks {
+		for _, attachment := range disk.Attachments {
+			fmt.Fprintf(tw, "%s\t%t\t%s\t%s\t%s\t%s\t%d\n",
+				attachment.Tag,
+				attachment.Attached,
+				attachment.MachineId,
+				attachment.DeviceName,
+				attachment.UUID,
+				attachment.Label,
+				attachment.Size,
```

Comment:
Please use go-humanize for the size (see my changes to "storage list"). I think we should only do this for the tabular format; the other formats are more likely to be used programmatically.